android/app/src/main/java/org/fossasia/openevent/viewmodels/TracksFragmentViewModel.java

```diff
@@ -0,0 +1,50 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.Transformations;
+import android.arch.lifecycle.ViewModel;
+
+import org.fossasia.openevent.data.Track;
+import org.fossasia.openevent.dbutils.FilterableRealmLiveData;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+
+import java.util.List;
+import java.util.Locale;
+
+import io.reactivex.functions.Predicate;
+
+public class TracksFragmentViewModel extends ViewModel {
+
+    private LiveData<List<Track>> tracksList;
+    private FilterableRealmLiveData<Track> filterableRealmLiveData;
+
+    private String searchText = "";
+
+    public TracksFragmentViewModel() {
+        filterableRealmLiveData = RealmDataRepository.asFilterableLiveData(RealmDataRepository.getDefaultInstance().getTracks());
+    }
+
+    public LiveData<List<Track>> getTracks(String searchText) {
+        if (!this.searchText.equals(searchText) || tracksList == null) {
+            setSearchText(searchText);
+            final String query = searchText.toLowerCase(Locale.getDefault());
+            Predicate<Track> predicate = track -> track.getName()
+                    .toLowerCase(Locale.getDefault())
+                    .contains(query);
+            filterableRealmLiveData.filter(predicate);
+            if (tracksList == null) {
+                tracksList = Transformations.map(filterableRealmLiveData, input -> input);
+            }
```

Comment:
FilterableRealmLiveData is already LiveData of List right, why do we need to transform again?

Reply:
Yes, it was redundant.Removed it, please review